pkg/daemon/server/service/pipeline_metrics_query_service.go

```diff
@@ -0,0 +1,192 @@
+package service
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/common/expfmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	"github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/apis/proto/daemon"
+	"github.com/numaproj/numaflow/pkg/isbsvc"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+)
+
+// metricsHttpClient interface for the GET call to metrics endpoint.
+// Had to add this an interface for testing
+type metricsHttpClient interface {
+	Get(url string) (*http.Response, error)
+}
+
+type pipelineMetricsQueryService struct {
+	isbsvcClient isbsvc.ISBService
+	pipeline     *v1alpha1.Pipeline
+	httpClient   metricsHttpClient
+}
+
+// NewPipelineMetricsQueryService returns a new instance of pipelineMetricsQueryService
+func NewPipelineMetricsQueryService(isbSvcClient isbsvc.ISBService, pipeline *v1alpha1.Pipeline) *pipelineMetricsQueryService {
+	return &pipelineMetricsQueryService{
+		isbsvcClient: isbSvcClient,
+		pipeline:     pipeline,
+		httpClient: &http.Client{Transport: &http.Transport{
```

Comment:
A short timeout (in few seconds) might be better (default timeout is 0, which is no timeout).

Reply:
Done